openmc/mesh.py

```diff
@@ -624,6 +628,333 @@ def to_xml_element(self):
 
         return element
 
+class CylindricalMesh(MeshBase):
+    """A 3D cylindrical mesh
+
+    Parameters
+    ----------
+    mesh_id : int
+        Unique identifier for the mesh
+    name : str
+        Name of the mesh
+
+    Attributes
+    ----------
+    id : int
+        Unique identifier for the mesh
+    name : str
+        Name of the mesh
+    dimension : Iterable of int
+        The number of mesh cells in each direction.
+    n_dimension : int
+        Number of mesh dimensions (always 3 for a CylindricalMesh).
+    r_grid : Iterable of float
+        Mesh boundary points along the r-axis.
+    phi_grid : Iterable of float
+        Mesh boundary points along the phi-axis.
```

Comment:
A mention of the default value for the phi grid (`(0, 360)`) would be good here.